File: SpringTILCode/src/main/java/com/example/springtilcode/domain/user/service/UserService.java
package com.example.springtilcode.domain.user.service;

import com.example.springtilcode.domain.user.controller.dto.UserResponse;
import com.example.springtilcode.domain.user.entity.User;
import com.example.springtilcode.domain.user.exception.UserNotFoundException;
import com.example.springtilcode.domain.user.repository.UserRepository;
import com.example.springtilcode.global.utils.SecurityUtil;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import javax.transaction.Transactional;

@Service
@RequiredArgsConstructor
public class UserService {

    private final UserRepository userRepository;

    @Transactional
    public UserResponse getUser(){
        User user = userRepository.findByEmail(SecurityUtil.getEmail()).orElseThrow(UserNotFoundException::new);
        return UserResponse.builder()
                .id(user.getId())
                .email(user.getEmail())
                .postList(user.getPostList())
                .build();
    }

    @Transactional
    public UserResponse editProfile(String email){
        User user = userRepository.findByEmail(SecurityUtil.getEmail()).orElseThrow(UserNotFoundException::new);
        user.editProfile(email);
        userRepository.save(user);

        return UserResponse.builder()
                .id(user.getId())
                .email(user.getEmail())
                .postList(user.getPostList())
                .build();
    }
}

File: kotlin/src/main/kotlin/com/example/demo/domain/user/service/UserService.kt
package com.example.demo.domain.user.service

import com.example.demo.domain.user.api.dto.response.UserResponse
import com.example.demo.domain.user.exception.AlreadyExistException
import com.example.demo.domain.user.exception.EmailNotFoundException
import com.example.demo.domain.user.repository.UserRepository
import com.example.demo.global.error.GlobalErrorCode
import com.example.demo.global.util.SecurityUtil
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class UserService (
    val userRepository: UserRepository,
    val securityUtil: SecurityUtil
){
    @Transactional
    fun getUser() : UserResponse {
        val user = userRepository.findByEmail(securityUtil.getEmail())
            ?: throw EmailNotFoundException(GlobalErrorCode.EMAIL_NOT_FOUND)
        return UserResponse(user.email, user.postList)
    }

    @Transactional
    fun editProfile(email: String): UserResponse {
        if(userRepository.findByEmail(email)?.equals(null) == false)
            throw AlreadyExistException(GlobalErrorCode.ALREADY_EXIST)
        val user = userRepository.findByEmail(securityUtil.getEmail())
            ?: throw EmailNotFoundException(GlobalErrorCode.EMAIL_NOT_FOUND)
        user.editProfile(email)
        return UserResponse(user.email, user.postList)
    }
}